crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -20,6 +21,90 @@ where
         unimplemented!()
     }
 
+    /// Finds a given trasaction by its hash.
+    pub(crate) async fn transaction_by_hash(
+        &self,
+        hash: H256,
+    ) -> EthResult<Option<reth_rpc_types::Transaction>> {
+        let tx_by_hash = match self.client().transaction_by_hash(hash) {
+            Ok(Some(item)) => item,
+            Ok(None) => return Ok(None),
+            Err(_) => return Err(EthApiError::EmptyRawTransactionData),
+        };
+
+        let tx: reth_rpc_types::Transaction = Transaction::from_recovered_with_block_context(
+            tx_by_hash.into_ecrecovered().unwrap(),
```

Comment:
no unwraps please.
this shouled be an `EthApiError::InvalidTransactionSignature` error

Reply:
fixed